### PR TITLE
[text analytics] Analyze updates for v5.1.0b6

### PR DESCRIPTION
**Pull request by abhahn**
fixes #16372

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py [hunk 1]
```diff
@@ -1382,12 +1383,49 @@ def __repr__(self, **kwargs):
...
+        self.string_index_type = kwargs.get("string_index_type", "UnicodeCodePoint")
+
+    def __repr__(self, **kwargs):
+        return "RecognizeLinkedEntitiesAction(model_version={}, string_index_type={})" \
+            .format(self.model_version, self.string_index_type)[:1024]
+
+    def to_generated(self):
+        return _latest_models.EntityLinkingTask(
+            parameters=_latest_models.EntityLinkingTaskParameters(
+                model_version=self.model_version
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py:1424]: I think `string_index_type` should be passed as well

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py [hunk 2]
```diff
@@ -804,6 +805,10 @@ def begin_analyze_batch_actions(  # type: ignore
                 key_phrase_extraction_tasks=[
                     t.to_generated() for t in
                     [a for a in actions if _determine_action_type(a) == AnalyzeBatchActionsType.EXTRACT_KEY_PHRASES]
+                ],
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py:808]: can you update the typehint on line 734?

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/_operations_mixin.py [hunk 3]
```diff
@@ -54,13 +54,11 @@ def analyze_status(
         :type skip: int
         :keyword callable cls: A custom type or function that will be passed the direct response
         :return: AnalyzeJobState, or the result of cls(response)
-        :rtype: ~azure.ai.textanalytics.v3_1_preview_3.models.AnalyzeJobState
+        :rtype: ~azure.ai.textanalytics.v3_1_preview_4.models.AnalyzeJobState
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/_operations_mixin.py:57]: can you add a changelog for this addition?

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py [hunk 4]
```diff
@@ -789,6 +790,10 @@ async def begin_analyze_batch_actions(  # type: ignore
                 key_phrase_extraction_tasks=[
                     t.to_generated() for t in
                     [a for a in actions if _determine_action_type(a) == AnalyzeBatchActionsType.EXTRACT_KEY_PHRASES]
+                ],
+                entity_linking_tasks=[
+                    t.to_generated() for t in
+                    [a for a in actions if _determine_action_type(a) == AnalyzeBatchActionsType.RECOGNIZE_LINKED_ENTITIES]
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py:796]: can you add some tests for linked entities and update the samples (readme and actual samples)? thanks!

Comment (abhahn) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py:796]: Yeah, I was working on this, just ran out of time yesterday.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py (hunk 1 above)
Comment (abhahn) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py]: Good catch, thanks!  I've updated this.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py (hunk 2 above)
Comment (abhahn) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py:808]: Updated.  I also had to disable `line-too-long` for pylint on this one.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/_operations_mixin.py (hunk 3 above)
Comment (abhahn) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/_operations_mixin.py:57]: Added.  Let me know if there's anything missing there.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py (hunk 4 above)
Comment (abhahn) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py:796]: Ok, I've updated tests and samples.  I've deleted a few recordings in preparation for running them, but I don't think this has been deployed to WestUS2 yet so I will wait to re-run the tests for now.  In the meantime let me know if there are any other tests I should modify/add.

**Review by iscai-msft — APPROVED** [review]

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/aio/_operations_mixin.py [hunk 5]
```diff
@@ -139,7 +139,7 @@ async def begin_health(
         self,
         documents: List["_models.MultiLanguageInput"],
         model_version: Optional[str] = None,
-        string_index_type: Optional[Union[str, "_models.StringIndexType"]] = "TextElements_v8",
+        string_index_type: Optional[Union[str, "_models.StringIndexType"]] = None,
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/aio/_operations_mixin.py:142]: @abhan why is the default here `None`, it should still be `TextElements_v8`

**Review by iscai-msft — APPROVED** [review]
barring the diff in the generated code, rest looks good. thanks @abhahn !

**Pull request merged**